src/test/basic.ts

```diff
@@ -1,195 +1,204 @@
-    assert(contextDoneSpy.called, 'context.done should be called')
-
-    const body = context.res.body
-
-    assert(typeof body === 'object', 'context.res.body should exist as an object')
-    assert(body.hasOwnProperty('__audits'), 'body contains __audits property')
-    assert(body.__audits[0].extensionLog[0] === 'Hello, World!', 'log item 1 is added to audit correctly')
-  })
-})
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+
+import isEqual = require('lodash.isequal')
+
+import { IAudit } from '../audit'
+import index from '../index'
+
+interface IHeader {
+  Authorization?: string
+}
+
+interface IBody {
+  __audits?: IAudit[]
+}
+
+describe('platform-chaos', () => {
+  it('is named properly', () => {
+    assert.equal(require('../../package.json').name, 'platform-chaos')
+    assert.equal(require('../../package-lock.json').name, 'platform-chaos')
+  })
+  it('parses resources', () => {
+    assert.throws(() => {
+      index.validators.resources({
+        body: {
+          resources: [1]
+        }
+      })
+    })
+
+    assert.throws(() => {
+      index.validators.resources({
+        body: {
+          resources: ['']
+        }
+      })
+    })
+
+    assert.throws(() => {
+      index.validators.resources({
+        body: {
+          resources: ['one/two/three/four']
+        }
+      })
+    })
+
+    const instance = index.parsers.resourcesToObjects({
+      body: {
+        resources: ['sub/rg/resource']
+      }
+    })[0]
+
+    assert.equal(instance.subscriptionId, 'sub')
+    assert.equal(instance.resourceGroupName, 'rg')
+    assert.equal(instance.resourceName, 'resource')
+  })
+
+  it('parses accessTokens', () => {
+    // invalid at
+    assert.throws(() => {
+      index.validators.accessToken({
+        body: {
+          accessToken: 1
+        }
+      })
+    })
+
+    // empty resourceIds
+    assert.throws(() => {
+      index.validators.accessToken({
+        body: {
+          accessToken: 'valid type'
+        }
+      })
+    })
+
+    // invalid resourceIds type
+    assert.throws(() => {
+      index.validators.accessToken({
+        body: {
+          accessToken: 'valid type'
+        }
+      })
+    })
+
+    // invalid resourceIds format
+    assert.throws(() => {
+      index.validators.accessToken({
+        body: {
+          accessToken: 'valid type'
+        }
+      })
+    })
+
+    // valid
+    const expectedAccessToken = 'Bearer 12345234r2'
+    const instance = index.parsers.accessTokenToCredentials({
+      body: {
+        accessToken: expectedAccessToken
+      }
+    })
+
+    const res = {
+      headers: {} as IHeader
+    }
+
+    instance.signRequest(res, () => null)
+
+    assert.equal(res.headers.Authorization, expectedAccessToken)
+  })
+
+  it('audits correctly', () => {
+    function contextLog () {
+      // in reality this would be `console.log(...arguments)`
+      // but in order to not cluter test we noop this

```

Comment:
[nit] `clutter`

Reply:
i think this is a relevant comment so that users who look at the test code to see how to use the auditor will know why we made this coding decision. if you think otherwise lmk and i'll remove it 👍

Reply:
no like clutter is spelled wrong

Reply:
lol @ this interaction ✨ ❤️

Reply:
LOL